test/util/xml_utils_unit.js

```diff
@@ -371,5 +371,69 @@ describe('XmlUtils', () => {
     expect(XmlUtils.parseFloat(HUGE_NUMBER_STRING)).toBe(Infinity);
     expect(XmlUtils.parseFloat('-' + HUGE_NUMBER_STRING)).toBe(-Infinity);
   });
+
+  describe('parseXmlString', () => {
```

Comment:
I wonder if it would be possible to add a `require-trusted-types-for` meta-tag to the test runner, so that our nightly testing verifies our XSS security...

I guess even if it was, the code from the demo page would also need to be updated, which would be work far out of the scope of this PR.

Reply:
Yeah definitely! We've done this both for unit and integration tests in open source projects in the past. For Karma it should just be a matter of adding the following to `karma.conf.js`:
```
  config.set({
    customHeaders: [
      {
        match: '.*',
        name: 'content-security-policy-report-only',
        value: "require-trusted-types-for 'script'"
      }
    ]
  });
```
Unfortunately this seems to cause the test suite to fail to initialize for some reason (we should have fixed all the Karma-specific Trusted Types issues by now, so I'm not sure why). A caveat about this is that you'll also need to address Trusted Types violations that are in the unit testing code, not only violations that come from the actual library. So setting this up in integration tests (the demo page?) would likely be a more suitable place. Based on static code analysis and exploring Trusted Types violations in applications that use (albeit perhaps only a small part of) Shaka Player, the only violation I'm aware of is the one coming from DOMParser, which will be fixed as part of this PR.